### PR TITLE
br: Ignore ddl jobs with empty query or blacklist type when exec restore

### LINKED ISSUE #33322: incremental restore doesn't empty query job.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
It's hard to reproduce it but easy to fix.
because we use `ExecuteInternal` to Execute DDL job queries during incremental restore.
and sometimes this query might be nil. so the [check](https://github.com/pingcap/tidb/blob/9f72bcded79688a9f50f07f4cc76889c534bd633/session/session.go#L1601-L1603) will not pass.

To Fix it we can ignore the empty job.Query in https://github.com/pingcap/tidb/blob/49f291db70a466c1a05718f8068b5234f70db1da/br/pkg/restore/client.go#L750

### 2. What did you expect to see? (Required)
incremental restore succeed.
### 3. What did you see instead (Required)
```
[2022/03/22 11:04:16.790 +08:00] [ERROR] [restore.go:35] ["failed to restore"] [error="[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use run multiple statements internally is not supported"]
```
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by WangLe1321**
### What problem does this PR solve?

Issue Number: close #33322

Problem Summary: Sometimes, query of ddl job can be empty, which will cause error when execute incremental restore.

### What is changed and how it works?
During backup, when we find ddl job with action types about TiFlash or lock/unlock table, we will set the query empty. So we add an action type blacklist, ddl job with action type in blacklist will be ignored. We also ignore ddl job with empty query in case other unexpected situation.

This means that we will ignore ddl jobs about TiFlash or lock/unlock table and ddl jobs with empty query, so they won't cause error by mistake.

### Check List

- [x] Unit test

### Release note

```release-note
Fix a bug that BR incremental restore return error by mistake caused by ddl job with empty query.
```

### CODE REVIEW

File: br/pkg/restore/db.go [hunk 1]
```diff
@@ -90,6 +90,11 @@ func (db *DB) ExecDDL(ctx context.Context, ddlJob *model.Job) error {
 		return errors.Trace(err)
 	}
 
+	// If query of the ddl job is empty or type of ddl job is in blacklist, ignore the ddl job.
+	if ddlJob.Query == "" || checkIsInActionList(ddlJob.Type, incrementalRestoreActionBlacklist) {
```
Comment (joccau) [inline br/pkg/restore/db.go:94]: `ExecDDL()` should be a universal interface,  don't check `checkIsInActionList()` in this function, I think.
Maybe put `checkIsInActionList()` in `FilterDDLJobs()`, or create a new function after  `FilterDDLJobs()`;
https://github.com/pingcap/tidb/blob/5a14cc3fd69c96a4a9c2348e4e87df7e49ae2ec1/br/pkg/restore/db.go#L361

**Review by joccau — CHANGES_REQUESTED** [review]

File: br/pkg/restore/db.go (hunk 1 above)
Comment (joccau) [inline br/pkg/restore/db.go:94]: `ExecDDL()` should be a universal interface,  don't check `checkIsInActionList()` in this function, I think.
Maybe put `checkIsInActionList()` in `FilterDDLJobs()`, or create a new function after  `FilterDDLJobs()`;
https://github.com/pingcap/tidb/blob/5a14cc3fd69c96a4a9c2348e4e87df7e49ae2ec1/br/pkg/restore/db.go#L361

File: br/pkg/restore/config.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package restore
```
Comment (3pointer) [inline br/pkg/restore/config.go:1]: please add license  here

File: br/pkg/restore/config.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+import "github.com/pingcap/tidb/parser/model"
+
+var incrementalRestoreActionBlacklist = []model.ActionType{
```
Comment (3pointer) [inline br/pkg/restore/config.go:5]: if this file only contains this variable, I think you can put this variable and function in same file.

**Review by joccau — APPROVED** [review]
LGTM.

File: br/pkg/restore/db.go [hunk 4]
```diff
@@ -90,6 +99,10 @@ func (db *DB) ExecDDL(ctx context.Context, ddlJob *model.Job) error {
 		return errors.Trace(err)
 	}
 
+	if ddlJob.Query == "" {
+		return nil
```
Comment (3pointer) [inline br/pkg/restore/db.go:103]: please add a warn log to record this DDL. something like DDL type

File: br/pkg/restore/db.go [hunk 5]
```diff
@@ -409,6 +422,31 @@ func FilterDDLJobs(allDDLJobs []*model.Job, tables []*metautil.Table) (ddlJobs [
...
+		if passed {
+			dstDDLJobs = append(dstDDLJobs, ddlJob)
+		}
+	}
+
+	return
+}
+
+// DDLJobBlacklistRule rule for filter ddl job with type in blacklist.
+func DDLJobBlacklistRule(ddlJob *model.Job) bool {
```
Comment (3pointer) [inline br/pkg/restore/db.go:446]: DDLJobBlacklistRule -> DDLJobBlockListRule

**Review by 3pointer — COMMENTED** [review]
rest LGTM

**Comment by fengou1** [comment]
LGTM.

File: br/pkg/restore/db.go (hunk 4 above)
Comment (WangLe1321) [reply br/pkg/restore/db.go:106]: done

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**